Source/Core/DolphinWX/ISOFile.cpp

```diff
@@ -236,10 +244,12 @@ std::string GameListItem::GetName() const
 {
 	bool wii = m_Platform != DiscIO::IVolume::GAMECUBE_DISC;
 	std::string name = GetName(SConfig::GetInstance().GetCurrentLanguage(wii));
+	std::string ext;
 	if (name.empty())
 	{
 		// No usable name, return filename (better than nothing)
-		SplitPath(GetFileName(), nullptr, &name, nullptr);
+		SplitPath(GetFileName(), nullptr, &name, &ext);
+		return name + ext;
```

Comment:
Won't this get rid of the `.` that the extension starts with? As an alternative, you could make a `wxFileName` object and call `GetFullName()` on it.